posts/boost-your-databricks-jobs-and-queries-performance.md

```diff
@@ -0,0 +1,66 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Data
+- Databricks
+- ETL
+- Performance
+date: 2023-02-18T16:09:09.000Z
+title: "Boost Your Databricks Jobs And Queries Performance"
+image: 
+---
+
+Databricks is doing a lot of optimization and caching by default to have jobs and queries run fast enough. Poorly designed tables (like having too many small partitions) or using a small cluster will obviously impact your performance. To avoid that, I recommend paying attention to those details when creating tables or clusters. In this blog, I will explain multiple techniques that can be done to enhance what Databricks is doing under the hood.
+
+# A fixed-size cluster is usually better than autoscaling
+
+Having a cluster that autoscales is great if you want to save money and if you don't have strict SLAs on your jobs/queries runtime. But you will realize that it is usually better to use a cluster with a fixed-size, especially if you know the volume of data being processed in advance. Upsizing your cluster is not instant and will usually take 3-4 minutes to have new worker nodes ready because of their initialization script.
```

Comment:
```suggestion
Having a cluster that autoscales is great if you want to save money and if you don't have strict SLAs on your job/query runtime. But you will realize that it is usually better to use a cluster with a fixed-size, especially if you know the volume of data being processed in advance. Upsizing your cluster is not instant and will usually take 3-4 minutes to have new worker nodes ready because of their initialization script.
```